File: frontend/src/schemas.ts
import { z } from 'zod';

export const GuestSchema = z.object({
  id: z.number(),
  group_id: z.number(),
  name: z.string(),
  created_at: z.string(),
});

export const GroupSchema = z.object({
  id: z.number(),
  name: z.string(),
  invited_to_nikkah: z.boolean(),
  invited_to_wedding: z.boolean(),
  invited_to_henna: z.boolean(),
  max_guests_wedding: z.number(),
  max_guests_henna: z.number(),
  has_accepted_wedding: z.boolean(),
  has_accepted_henna: z.boolean(),
  has_rsvped_wedding: z.boolean(),
  has_rsvped_henna: z.boolean(),
  wedding_guests: z.array(GuestSchema),
  henna_guests: z.array(GuestSchema),
});

export const RSVPRequestSchema = z.object({
  event: z.enum(['wedding', 'henna']),
  accept: z.boolean(),
  guests: z.array(z.string()),
});

export const RSVPResponseSchema = z.object({
  success: z.boolean(),
  message: z.string(),
  group: GroupSchema.optional(),
});

export type Guest = z.infer<typeof GuestSchema>;
export type Group = z.infer<typeof GroupSchema>;
export type RSVPRequest = z.infer<typeof RSVPRequestSchema>;
export type RSVPResponse = z.infer<typeof RSVPResponseSchema>;

File: backend/src/schemas.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime


class GuestBase(BaseModel):
    name: str


class GuestCreate(GuestBase):
    pass


class WeddingGuest(GuestBase):
    id: int
    group_id: int
    created_at: datetime

    class Config:
        from_attributes = True


class HennaGuest(GuestBase):
    id: int
    group_id: int
    created_at: datetime

    class Config:
        from_attributes = True


class GroupBase(BaseModel):
    name: str


class GroupResponse(GroupBase):
    id: int
    invited_to_nikkah: bool
    invited_to_wedding: bool
    invited_to_henna: bool
    max_guests_wedding: int
    max_guests_henna: int
    has_accepted_wedding: bool
    has_accepted_henna: bool
    has_rsvped_wedding: bool
    has_rsvped_henna: bool
    wedding_guests: List[WeddingGuest] = []
    henna_guests: List[HennaGuest] = []

    class Config:
        from_attributes = True


class RSVPRequest(BaseModel):
    event: str  # "wedding" or "henna"
    accept: bool
    guests: List[str] = []


class RSVPResponse(BaseModel):
    success: bool
    message: str
    group: Optional[GroupResponse] = None
